src/Microsoft.AspNet.Razor/CodeGenerators/CSharpLineMappingWriter.cs

```diff
@@ -114,7 +114,7 @@ public void Dispose()
             {
                 // Need to add an additional line at the end IF there wasn't one already written.
                 // This is needed to work with the C# editor's handling of #line ...
-                var endsWithNewline = _writer.GenerateCode().EndsWith("\n");
```

Comment:
`StringBuilder` doesn't have an EndsWith method. I think @ellismg was referring to the string we were producing. I added a bounds check for the char check.